pkg/gateway/services/stub.go

```diff
@@ -21,6 +21,23 @@ func (gws *GatewayService) GetOrCreateStub(ctx context.Context, in *pb.GetOrCrea
 		}, nil
 	}
 
+	if in.Gpu != "" {
+		concurrencyLimit, err := gws.backendRepo.GetConcurrencyLimitByWorkspaceId(ctx, authInfo.Workspace.ExternalId)
+		if err != nil {
+			return &pb.GetOrCreateStubResponse{
+				Ok:     false,
+				ErrMsg: "Failed to get concurrency limit for workspace",
+			}, nil
+		}
+
+		if concurrencyLimit.GPULimit <= 0 {
+			return &pb.GetOrCreateStubResponse{
+				Ok:     false,
+				ErrMsg: "GPU concurrency limit is 0. A payment method is required to deploy with GPUs.",
```

Comment:
Since this in beta9, and there's no concept of "payment", I would just let the error be the first part of that message.